pkg/controllers/sandbox/unsandboxsubcluster_reconciler.go

```diff
@@ -126,11 +126,15 @@ func (r *UnsandboxSubclusterReconciler) reconcileSandboxInfoInVdb(ctx context.Co
 // reconcileSandboxConfigMap will update/delete sandbox config map if it expires, this function will return
 // an error and a boolean to indicate if sandbox config map is deleted
 func (r *UnsandboxSubclusterReconciler) reconcileSandboxConfigMap(ctx context.Context) (error, bool) {
+	if err := r.OriginalPFacts.Collect(ctx, r.Vdb); err != nil {
```

Comment:
Maybe this is a sign that reconcileConfigMap should be in a separate reconciler.

Reply:
I do have it in vdbcontroller.sandboxsubcluster_reconciler. But I want to keep one in sandbox controller too. The reconcilers inside sandbox controller will all have this issue. In the future, we can improve the method we used to udpate vdb: after we update vdb, we should poll/assign the vdb changes to the VDB of reconciler.